### PR TITLE
build: bundle the library in angular package format

### PR DESCRIPTION
**Pull request by mattlewis92**
I was trying out our app with the new ivy compiler and this package wasn't working with it because it wasn't in the angular package format, this refactors all the build process to use ng-packagr which nicely takes care of all the hard work. Thanks for your work on this library! 😄

### CODE REVIEW

**Comment by CunningFatalist** [comment]
Hello @mattlewis92! 

Thanks a lot for your pull request, it is very much appreciated! Our company is mostly using Vue nowadays, and I am not familiar with the official Angular package format. Therefore, your contribution is more than welcome. 

There are some things that I have to find out/make sure before I can merge this pull request, though.

- How does this affect users of older Angular versions (2 + 4)?
- What is the exact difference to the current output?
- Do your changes actually mind the difference between `README.md` (GitHub) and `README_NPM.md` (NPM)?

```
.
├── README.md
├── cookie-service
│   ├── cookie.service.d.ts
│   ├── cookie.service.js
│   ├── cookie.service.js.map
│   ├── cookie.service.metadata.json
│   └── cookie.service.ngsummary.json
├── index.d.ts
├── index.js
├── index.js.map
├── index.metadata.json
├── index.ngsummary.json
└── package.json
```

I will find these things out soon(ish) and merge your request afterwards. In the meantime, feel free to add yourself to the list of contributors (see `package.json` and `README.md`).

Cheers from Berlin 😃

**Comment by mattlewis92** [comment]
> How does this affect users of older Angular versions (2 + 4)?

It doesn't change anything for those users. It just bundles the library in different ways, it's still consumed exactly the same way.

> Do your changes actually mind the difference between README.md (GitHub) and README_NPM.md (NPM)?

ng-packagr copies the README.md into the dist folder, so it would just ignore the `README_NPM.md`. I diff'd them and the 2 files are the same so it seemed safe to delete the duplicate.

> What is the exact difference to the current output?

The angular package format bundles the library in different formats, either in es2015 format or es5 with es modules or as a UMD bundle. You can take a look at [one of my libraries](https://unpkg.com/angular-draggable-droppable@4.3.0/) that uses the format to see how it gets outputted.

**Comment by CunningFatalist** [comment]
Hey,

thanks for your fast reply. Also, thanks for sharing the example output of your bundled library, that helped a bunch. [README.md](https://github.com/7leads/ngx-cookie-service/blob/master/README.md) and [README_NPM.md](https://github.com/7leads/ngx-cookie-service/blob/master/README_NPM.md) are different, though.

Currently, I can see two ways of handling this.

1. Making your process work with both readme files.
2. Deleting the extra NPM readme file.

I think it might be best to just delete the additional readme file. The readme changed over time, and I do not see any additional value in having two of them anymore. I have to talk to my colleague first, though, and will write you again soon.

Cheeers

**Comment by mattlewis92** [comment]
Oh weird, when I diffed them, I could have sworn they were the same. I defo think deleting the extra one is best though, as they'll just get out of sync and I can't really see any advantage to having a separate one.

**Comment by CunningFatalist** [comment]
Hello,

sorry for the late answer! We discussed this, and you are right. `README_NPM.md` does not make sense anymore. Thanks for deleting it.

However, there is still one problem left with this PR. I pulled your branch and tried `yarn compile`, but I get the error "`public_api.ts` not found". This makes sense, as it does not exist. I am not familiar with `ng-packagr`, but I think there are two possible solutions. Either add the `public_api.ts` file, or add `ngPackage` options to `package.json`. To me, it looks like you might have `ng-package.json` on your machine (I think so because of the script `"compile": "ng-packagr -p ng-package.json"` in `package.json`). Maybe you forgot to commit this file?

So, in summary the only things left to do are:

- Add yourself to the list of contributors in `package.json` and `README.md`
- Fix the compile issue

Thank you so much for your contribution. I know a lot of people will appreciate it 👍 

Cheers

**Comment by mattlewis92** [comment]
Ahh, it's because of the gitignore that the file wasn't added, fixed now! Thanks for reviewing 😄

**Comment by CunningFatalist** [comment]
Hello @mattlewis92, 

I am very sorry for the late response (yet again). I was quite busy and did not have the time to test everything probably. Everything looks great to me, thank you so much for your contribution 👍 

Cheers

**Comment by mattlewis92** [comment]
Awesome, thanks! 😄
